ush/ioda/bufr2ioda/bufr2ioda_conventional_prepbufr_ps.py

```diff
@@ -39,6 +39,33 @@ def Mask_typ_for_var(typ, var):
     return typ_var
 
 
+def Compute_obsSubType(typ, t29, sid):
+
+    obssubtype = ma.array(np.full(typ.shape[0], 0))
+    for i in range(len(typ)):
+        newtyp = typ[i]
+        if ((typ[i] == 180) or (typ[i] == 280)) and (t29[i] == 562):
+            newval = 0
+            try:
+                newval = int(sid[i][-3:])
+            except ValueError:
+                newval = 0
+
+            if newval > 500:
+                newtyp == typ[i] + 19
+
+        if ((typ[i] == 180) and ((t29[i] == 522) or (t29[i] == 523))):
+            newtyp == typ[i] + 18
```

Comment:
@nicholasesposito for conventional stationPressure application, we do not need to create new types for bouy and ship.  This is specific to the NSST application (where the id_drifter and id_ship are set to true).   

This is my bad.  I did not read the GSI code carefully.  
```
        if (id_drifter .and. (kx==180 .or. kx==280) .and. nint(hdr(3),r_double)==562) then
           rstation_id=hdr(4)
           read(c_station_id,*,iostat=ios) iwmo
           if (ios == 0 .and. iwmo > 0) then
              if(mod(iwmo,1000) >=500) then
                 kx = kx + 19
              end if
           end if
        end if

        if (id_ship .and. (kx==180) .and. (nint(hdr(3),r_double)==522 .or. nint(hdr(3),r_double)==523)) then
           rstation_id=hdr(4)
           kx = kx + 18
        end if
```

The code you spotted in GSI and already added to this PR is what we want.  
```
! temporary specify iobsub until put in bufr file
        iobsub = 0
        if(kx == 280 .or. kx == 180 ) iobsub=hdr(3)
        if(kx == 280 .or. kx ==180) then
          if ( hdr(3) >555.0_r_kind .and. hdr(3) <565.0_r_kind ) then
            iobsub=00
          else
            iobsub=01
          endif
        endif

```

Reply:
@emilyhcliu The way I see it, we still need the initial part of the code to "filter out" other typ's/kx's and add 18/19 to them where typ/kx has those specific conditions because then the second part works on everything that remains.

Reply:
At some point, it seems like it reverts back to 180/280 but I can't tell where. There is no 198/199/299 in the outputs. I can show you what I mean tomorrow.

Reply:
@nicholasesposito the 199, 198, and 299 are for NSST only when user set id_drifter and id_ship to true in the namelist. For Non SST data set, the id_drifter and id_ship are set to false

Reply:
@emilyhcliu Oh I think I see what you're saying. Let me comment them out and see what the new results would be.

Reply:
Yep, you're right. I made the changes and checked them in

Reply:
I tested the updated `conventional_ps` IODA file in the end-to-end test.  The IODA input file worked great!  
The number that passed QC comparison between GSI and JEDI is 85378 and 85387.   This is close enough.  👍